api/v1alpha1/map_types.go

```diff
@@ -63,6 +63,40 @@ type MapSpec struct {
 	// +kubebuilder:default:=BINARY
 	// +optional
 	InMemoryFormat InMemoryFormatType `json:"inMemoryFormat,omitempty"`
+
+	// EntryListeners contains the configuration for the map-level or entry-based events listeners
+	// provided by the Hazelcast’s eventing framework.
+	// You can learn more at https://docs.hazelcast.com/imdg/latest/events/object-events.
+	// +optional
+	EntryListeners []EntryListenerConfiguration `json:"entryListeners,omitempty"`
+}
+
+type EntryListenerConfiguration struct {
+	// ClassName is the fully qualified name of the class that implements any of the Listener interface.
+	// +kubebuilder:validation:MinLength:=1
+	ClassName string `json:"className"`
+
+	// IncludeValues is an optional attribute that indicates whether the event will contain the map value.
+	// +kubebuilder:default:=true
+	IncludeValues *bool `json:"includeValues"`
+
+	// Local is an optional attribute that indicates whether the map on the local member can be listened to.
+	// +kubebuilder:default:=false
+	Local *bool `json:"local"`
```

Comment:
If `local` is not optional, we do not need a default value in kubebuilder command. Should we make it optional?

Reply:
To keep it consistent with how we approach other fields, I would suggest we do something like this

```suggestion
        // +kubebuilder:default:=false
	Local bool `json:"local"`
```

I removed the pointer because the field has a default value and did not add omitempty tag because default go value and it real value are aligned. 

At the end I would suggest we delete `GetLocal()` function, and just use the field to get its value. WDYT?